### PR TITLE
i128 is being stabilized

### PR DESCRIPTION
**Pull request by mark-i-m**
cc rust-lang/rust#35118 rust-lang/rust#49101

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -12,7 +12,6 @@
 #![feature(compiler_builtins)]
 #![feature(core_intrinsics)]
 #![feature(naked_functions)]
-#![feature(i128_type)]
```
Comment (est31) [inline src/lib.rs]: You'd most likely have to do `#[cfg_attr(stage0, feature(i128_type))]` here, as on stage0 it is still needed.

Comment (mark-i-m) [reply src/lib.rs]: Hmm... It seemed to work fine when I made this change locally?

**Comment by alexcrichton** [comment]
Thanks! I think it's ok if this is here, right? Does this fail the build?

**Comment by alexcrichton** [comment]
(in that I'd prefer to land the rustc change first and then land the change here after that's gone through)

**Comment by PlasmaPower** [comment]
It's sort of a deadlock. Rustc fails without this, but this fails without rustc.

**Comment by alexcrichton** [comment]
The "failure" I think is `deny(warnings)`, right? Can that be removed?

**Comment by PlasmaPower** [comment]
Yeah I think that's the best solution. If we want we can re-add if after the rustc change goes through.

**Comment by mark-i-m** [comment]
Moving the `feature(i128_type)` after `allow(unused_features)` seems to make the compiler build for me locally.

**Comment by kennytm** [comment]
We could use the old external tool update workflow here:

1. Push this PR into a new, non-master branch on this repository 
2. Update the submodule in rust-lang/rust#35518 to point to that new branch
3. Wait for rust-lang/rust#35518 to merge and a new nightly is released
4. Merge this PR

**Comment by PlasmaPower** [comment]
That's what I originally suggested in the rustc PR but just ignoring the warning temporarily seems easier.

**Comment by mark-i-m** [comment]
Umm... does anyone understand the current Travis failure:

```
thread 'rustc' panicked at 'slice index starts at 4294967295 but ends at 30759529', libcore/slice/mod.rs:791:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: rustc 1.26.0-nightly (adf2135ad 2018-03-17) running on x86_64-apple-darwin
```

**Comment by mark-i-m** [comment]
It looks like the current change in the PR might be sufficient if we fix the build failures... alternately, I could do @kennytm's suggestion. Just let me know :)

**Comment by alexcrichton** [comment]
Ok this strategy seems fine by me yeah

**Comment by mark-i-m** [comment]
Hmm... this doesn't seem to have worked... I don't understand why.

File: src/lib.rs [hunk 2]
```diff
@@ -12,12 +12,12 @@
...
 #![feature(repr_simd)]
 #![feature(abi_unadjusted)]
 #![feature(linkage)]
 #![feature(lang_items)]
 #![allow(unused_features)]
```
Comment (PlasmaPower) [inline src/lib.rs]: Oh this is `#![allow(unused_features)]`, but you need `#[allow(stable_features)]`.

**Pull request merged**